core/src/Platforms/net45/UserPassCred.cs

```diff
@@ -0,0 +1,97 @@
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Microsoft.Identity.Core
+{
+    /// <summary>
+    /// Credential used for username/password authentication.
+    /// </summary>
+    public class UserPasswordCred : UserCred
+    {
+        /// <summary>
+        /// Constructor to create credential with username and password
+        /// </summary>
+        /// <param name="userName">Identifier of the user application requests token on behalf.</param>
+        /// <param name="password">User password.</param>
+        public UserPasswordCred(string userName, string password) : base(userName, UserAuthType.UsernamePassword)
+        {
+            this.Password = password;
+        }
+
+        /// <summary>
+        /// Constructor to create credential with username and password
+        /// </summary>
+        /// <param name="userName">Identifier of the user application requests token on behalf.</param>
+        /// <param name="securePassword">User password.</param>
+        public UserPasswordCred(string userName, SecureString securePassword) : base(userName, UserAuthType.UsernamePassword)
+        {
+            this.SecurePassword = securePassword;
+        }
+
+        internal SecureString SecurePassword { get; private set; }
+
+        internal string Password { get; }
+
+        internal override char[] PasswordToCharArray()
+        {
+            if (SecurePassword != null)
+            {
+                var output = new char[SecurePassword.Length];
+                IntPtr secureStringPtr = Marshal.SecureStringToCoTaskMemUnicode(SecurePassword);
+                for (int i = 0; i < SecurePassword.Length; i++)
+                {
+                    output[i] = (char)Marshal.ReadInt16(secureStringPtr, i * 2);
+                }
+
+                Marshal.ZeroFreeCoTaskMemUnicode(secureStringPtr);
```

Comment:
wondering if there exists a cross platform way of doing this instead of using net45 specifics.  also wonder if this section should be marked as unsafe?

Reply:
This line is copied from [existing implementation in ADAL](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118/files#diff-7b324a26d8acf2f7eb4a09d07dfd2513R78). We can defer this conversation to near future when we start to implement the username/password flow, since you are also asking to remove some other currently-commented `/* will be used by Username Password flow */` anyway.